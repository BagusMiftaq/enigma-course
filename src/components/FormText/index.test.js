import {render} from "@testing-library/react";

import FormText from "./index";

test("Should render FormText correctly", ()=>{
    const mockProps = {label: "this is label", type: "text", placeholder: "this is placeholder",
        value:{obj:""}, onChange: ()=>{}, disabled: false}

    const {asFragment} = render(<FormText {...mockProps}/>);

    expect(asFragment()).toMatchSnapshot();
})

test("Should render FormText correctly", ()=>{
    const mockProps = {label: "this is label", type: "textarea", placeholder: "this is placeholder",
        value:{obj:""}, onChange: ()=>{}, disabled: false}

    const {asFragment} = render(<FormText {...mockProps}/>);

    expect(asFragment()).toMatchSnapshot();
})