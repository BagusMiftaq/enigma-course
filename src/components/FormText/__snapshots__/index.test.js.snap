// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render FormText correctly 1`] = `
<DocumentFragment>
  <div
    class="mb-3"
  >
    <label
      class="form-label"
    >
      this is label
    </label>
    <input
      class="form-control is-valid"
      placeholder="this is placeholder"
      type="text"
      value="[object Object]"
    />
  </div>
</DocumentFragment>
`;

exports[`Should render FormText correctly 2`] = `
<DocumentFragment>
  <div
    class="mb-3"
  >
    <label
      class="form-label"
    >
      this is label
    </label>
    <textarea
      class="form-control is-valid"
      placeholder="this is placeholder"
    >
      [object Object]
    </textarea>
  </div>
</DocumentFragment>
`;
